### DISCUSSION TITLE
Failure running cmd in Custom Action on some computers

### DISCUSSION THREAD

**Discussion by shirborer-lls** [wixtoolset/issues#6630] [Questions]
Trying to add a custom action that will delete files from the installation folder using the following code:

    <CustomAction Id="InstallService" Return="check" Impersonate="no" Execute="deferred"
                              Directory="INSTALLFOLDER" ExeCommand='cmd.exe /C del /s /q "[INSTALLFOLDER]"'/>
     <InstallExecuteSequence>
          <Custom Action="InstallService" Before ="InstallFiles">NOT REMOVE</Custom>
     </InstallExecuteSequence>

This works on some computers but fails on others showing a message

    A program required for install to complete could not be run.

And in the installation logs I see

    Error 1721. There is a problem with this Windows Installer package. A program required for this install to complete could not be run. Contact your support personnel or package vendor

I've tried different value of Impersonate and Execute but nothing seemed to work.

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [experimental GitHub discussions area][gh]. You might also consider asking your question on the [wix-users mailing list][ml] or posting on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[ml]: http://wixtoolset.org/documentation/mailinglist/
[so]: https://stackoverflow.com/questions/tagged/wix

**Comment by robmen** [comment]
There should be a more specific error somewhere in the log file. Try searching for `value 3` as per https://robmensching.com/blog/posts/2010/8/2/the-first-thing-i-do-with-an-msi-log/.
